### ISSUE TITLE
Could not load type "checkbox": class does not exist.

### ISSUE THREAD

**Issue by franckentien** [sonata-project/SonataNewsBundle#431]
#### Sonata packages

```
sonata-project/admin-bundle              3.31.0 = 3.31.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.10.0 = 3.10.0 Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  = 2.0.1  Cache library
sonata-project/classification-bundle     3.6.0  = 3.6.0  Symfony SonataClassificationBundle
sonata-project/core-bundle               3.9.0  = 3.9.0  Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.3.1  = 2.3.1  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.0.2  = 1.0.2  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.4.1  = 3.4.1  Symfony Sonata / Integrate Doctrine ORM into the SonataAdm...
sonata-project/easy-extends-bundle       2.4.0  = 2.4.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.8.0  = 1.8.0  Lightweight Exporter library
sonata-project/formatter-bundle          3.4.0  = 3.4.0  Symfony SonataFormatterBundle
sonata-project/intl-bundle               2.4.0  = 2.4.0  Symfony SonataIntlBundle
sonata-project/media-bundle              3.10.0 = 3.10.0 Symfony SonataMediaBundle
sonata-project/news-bundle               3.2.2  = 3.2.2  Symfony SonataNewsBundle
sonata-project/notification-bundle       3.3.0  = 3.3.0  Symfony SonataNotificationBundle
sonata-project/user-bundle               4.1.0  = 4.1.0  Symfony SonataUserBundle
```

#### Symfony packages

```
symfony/assetic-bundle     v2.8.2 = v2.8.2 Integrates Assetic into Symfony2
symfony/monolog-bundle     v3.1.2 = v3.1.2 Symfony MonologBundle
symfony/phpunit-bridge     v3.4.3 ~ v4.0.3 Symfony PHPUnit Bridge
symfony/polyfill-apcu      v1.6.0 = v1.6.0 Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-intl-icu  v1.6.0 = v1.6.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.6.0 = v1.6.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php56     v1.6.0 = v1.6.0 Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php70     v1.6.0 = v1.6.0 Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-util      v1.6.0 = v1.6.0 Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.1 = v3.0.1 Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v2.6.7 ~ v3.1.6 Symfony SwiftmailerBundle
symfony/symfony            v3.4.3 ~ v4.0.3 The Symfony PHP framework
```

#### PHP version

```
PHP 7.1.9 (cli) (built: Aug 30 2017 18:34:46) ( ZTS MSVC14 (Visual C++ 2015) x64 )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
```

## Subject

With the most basique projet of Symfony with the AdminBundle, the NewsBundle and all bundles require.
I got an error when i wan to display the list of articles empty or not.

I have created an issue in Admin bundle but @greg0ire ask me to create it on this project.
https://github.com/sonata-project/SonataAdminBundle/issues/4903

## Steps to reproduce

Create an project, 
Add admin bundle, news bundle and all other bundle needed.
Display the admin page and click on 'List' of news.

## Expected results

The list of news or empty tab

## Actual results

Error page with : Could not load type "checkbox": class does not exist.

The stack trace says 'InvalidArgumentException'

The full Stack Trace:
```
Symfony\Component\Form\Exception\InvalidArgumentException:
Could not load type "checkbox": class does not exist.

  at vendor\symfony\symfony\src\Symfony\Component\Form\FormRegistry.php:86
  at Symfony\Component\Form\FormRegistry->getType('checkbox')
     (vendor\symfony\symfony\src\Symfony\Component\Form\FormFactory.php:74)
  at Symfony\Component\Form\FormFactory->createNamedBuilder('value', 'checkbox', null, array('required' => false))
     (vendor\symfony\symfony\src\Symfony\Component\Form\FormBuilder.php:106)
  at Symfony\Component\Form\FormBuilder->create('value', 'checkbox', array('required' => false))
     (vendor\symfony\symfony\src\Symfony\Component\Form\FormBuilder.php:269)
  at Symfony\Component\Form\FormBuilder->resolveChildren()
     (vendor\symfony\symfony\src\Symfony\Component\Form\FormBuilder.php:215)
  at Symfony\Component\Form\FormBuilder->getForm()
     (vendor\symfony\symfony\src\Symfony\Component\Form\FormBuilder.php:221)
  at Symfony\Component\Form\FormBuilder->getForm()
     (vendor\sonata-project\admin-bundle\src\Datagrid\Datagrid.php:133)
  at Sonata\AdminBundle\Datagrid\Datagrid->buildPager()
     (vendor\sonata-project\admin-bundle\src\Datagrid\Datagrid.php:259)
  at Sonata\AdminBundle\Datagrid\Datagrid->getForm()
     (vendor\sonata-project\admin-bundle\src\Controller\CRUDController.php:157)
  at Sonata\AdminBundle\Controller\CRUDController->listAction()
     (vendor\symfony\symfony\src\Symfony\Component\HttpKernel\HttpKernel.php:151)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor\symfony\symfony\src\Symfony\Component\HttpKernel\HttpKernel.php:68)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor\symfony\symfony\src\Symfony\Component\HttpKernel\Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web\app_dev.php:29)
```

**Comment by OskarStark** [comment]
please use the FQCN instead of `checkbox`

**Comment by franckentien** [comment]
How can i switch my checkbox option to FQNC ?

**Comment by greg0ire** [comment]
Please edit this line: https://github.com/sonata-project/SonataNewsBundle/blob/3.x/src/Admin/PostAdmin.php#L214

**Comment by greg0ire** [comment]
If it fixes your issue, please make a PR

**Comment by franckentien** [comment]
Edit the line by changing `ckeckbox` to `FQNC` do not solve the problem, same error.
But if I comment this line `'field_type' => 'checkbox',` it works.

Should i make a PR with this line in comment ?

Thanks

**Comment by greg0ire** [comment]
> Edit the line by changing ckeckbox to FQNC do not solve the problem, same error.

Please show exactly what you did, and the error message.

**Comment by franckentien** [comment]
I got the error on the url: /web/app_dev.php/admin/sonata/news/post/list

### Initial Error 

**Could not load type "checkbox": class does not exist.**

The error message: 
![image](https://user-images.githubusercontent.com/6872773/35328864-0c4a61fa-00fe-11e8-957c-ef100375cee2.png)

### FQNC Switch 

I have edited the line in `\vendor\sonata-project\news-bundle\src\Admin\PostAdmin.php` line 214 by changing `'field_type' => 'checkbox',` to `'field_type' => 'FQNC',`

I got this error: **Could not load type "FQNC": class does not exist.**

### Field type comment 

When i comment the line with the field_type parameter it work with no error

![image](https://user-images.githubusercontent.com/6872773/35329078-c9f5a3a4-00fe-11e8-8768-eaf88093334c.png)

I can give more information if you need

**Comment by greg0ire** [comment]
It's not FQNC, it's FQCN, which means "fully qualified class name", which means you need to put the FQCN of the checkbox type class, which `CheckboxType::class` will resolve to

TL;DR use `CheckboxType::class`

**Comment by franckentien** [comment]
Thanks, you have right.
I will make a PR with the modification.
The new line was: `'field_type' => CheckboxType::class,`

**Comment by greg0ire** [comment]
I asked for a release
